src/modules/commander/failsafe/failsafe.cpp

```diff
@@ -467,6 +467,7 @@ FailsafeBase::Action Failsafe::checkModeFallback(const failsafe_flags_s &status_
 	// offboard signal
 	if (status_flags.offboard_control_signal_lost && (status_flags.mode_req_offboard_signal & (1u << user_intended_mode))) {
 		action = fromOffboardLossActParam(_param_com_obl_rc_act.get(), user_intended_mode);
+		return action;
```

Comment:
Can you only return if the action is disarm, otherwise the posctrl -> altctrl won't be correct. Relying on multiple iterations isn't good.
`if (action == Action::Disarm) { return action; }`

Reply:
Sure